contracts/RegisterDomains.sol

```diff
@@ -0,0 +1,55 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.0;
+
+struct DomainDetails {
+    address controller;
+    uint deposit;
+}
+
+uint256 constant DEPOSIT_PRICE = 1 ether;
+
+contract RegisterDomains {
+    address payable public owner;
+    mapping(string => DomainDetails) domains;
+
+    constructor() {
+        owner = payable(msg.sender);
+    }
+
+    function getDomain(string memory domainName) external view returns (DomainDetails memory) {
+        return domains[domainName];
+    }
+
+    function createDomain(string memory domainName) internal {
+        DomainDetails memory domain = DomainDetails({
+            controller: owner,
+            deposit: msg.value
+        });
+        domains[domainName] = domain;
+    }
+
+    function deleteDomain(string memory domainName) internal {
+        delete domains[domainName];
+    }
+
+    function registerDomain(string memory domainName) external payable {
+        require(msg.value >= DEPOSIT_PRICE, "Insufficient ETH sent");
+
+        DomainDetails memory existingDomain = this.getDomain(domainName);
+        require(existingDomain.deposit == 0, "Domain is already reserved");
+
+        createDomain(domainName);
+    }
+
+    function unregisterDomain(string memory domainName) external payable {
+        DomainDetails memory existingDomain = this.getDomain(domainName);
```

Comment:
Is it mandatory to access via `this.` ? -1

Reply:
within this implementation yes its mandatory, compilator was failing because function was marked as `external`, updated it to `public` and removed `this`